tony-core/src/main/java/com/linkedin/tony/util/Utils.java

```diff
@@ -211,6 +210,15 @@ public static void setCapabilityGPU(Resource resource, int gpuCount) {
     return;
   }
 
+  /**
+   * Uses reflection to get memory size.
+   * @param resource  the request container resource
+   * @return the memory size
+   */
+  public static long getMemorySize(Resource resource) {
+    return HadoopCompatibleAdapter.getMemorySize(resource);
+  }
```

Comment:
remove these functions from utils? just call the compatible adapter from code is good enough.